clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -378,11 +378,19 @@ private void onErrorResponse(final R response, final long currentTimeMs) {
             case INVALID_REQUEST:
             case GROUP_MAX_SIZE_REACHED:
             case UNSUPPORTED_ASSIGNOR:
-            case UNSUPPORTED_VERSION:
                 logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, errorMessage);
                 handleFatalFailure(error.exception(errorMessage));
                 break;
 
+            case UNSUPPORTED_VERSION:
+                String customErrorMessage = "The cluster doesn't yet support the new consumer group protocol." +
+                        " Set group.protocol=classic to revert to the classic protocol until the cluster is upgraded.";
+                errorMessage = errorMessage == null || errorMessage.isEmpty() ? customErrorMessage : errorMessage + ". " + customErrorMessage;
```

Comment:
I wonder if we should just remove this. It will more or less always return the default error message so it won't look nice for the end user.